packages/peregrine/src/hooks/useCarousel.js

```diff
@@ -0,0 +1,28 @@
+import { useState, useCallback } from 'react';
+
+export const useCarousel = carouselSize => {
+    const [activeItemIndex, setActiveItemIndex] = useState(0);
+    const [currentImageLoaded, setCurrentImageLoaded] = useState(false);
+
+    const handleImageLoad = () => setCurrentImageLoaded(true);
+
+    const handlePrevious = useCallback(() =>
+        activeItemIndex > 0
+            ? setActiveItemIndex(activeItemIndex - 1)
+            : setActiveItemIndex(carouselSize - 1)
```

Comment:
```suggestion
            : setActiveItemIndex(carouselSize - 1), [activeItemIndex]
```
From the React Hooks API Reference for `useCallback`, 
> every value referenced inside the callback should also appear in the dependencies array

`carouselSize` is immutable but `activeItemIndex` will change, so we need to update the first `setActiveItemIndex` in the callback when `activeItemIndex` changes